File: src/main/Java/boj_step14/boj11478.java
import java.io.*;
import java.util.HashSet;

public class boj11478 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        String line = reader.readLine();
        HashSet<String> set = new HashSet<>();
        for (int i = 0; i < line.length(); i++)
            for (int j = i + 1; j <= line.length(); j++)
                set.add(line.substring(i, j));
        writer.append(String.valueOf(set.size())).close();
    }
}

File: src/main/kotlin/boj_step14/boj11478.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val str = r.readLine()
    val set = mutableSetOf<String>()
    repeat(str.length) {
        for (i in 0 until str.length - it) set.add(str.substring(i, i + it + 1))
    }
    w.write(set.size.toString())
    w.close()
}